### DISCUSSION TITLE
Termux With Or Without Root

### DISCUSSION THREAD

**Discussion by zxrovx** [termux/termux-app#2320] [Q&A] [answered]
Hi Can You Use Termux With Or Without Root?
So every time you install the application for the first time there is a notification that appears to choose to use root mode, or without root, or both.

**Comment by ghost** [comment]
Termux is primarily for non-rooted users. optional root packages is available in `root-repo` repository
and No Notification will be shown if you are rooted or not or what to choose because Termux is simply a non-rooted app

**Comment by ghost** [comment, marked as answer]
If your device is rooted, you can grant superuser (root) privileges by using `tsu` or `sudo` from package `tsu`. Root use is optional and application is primarily targeted at non-root use in general.

> So every time you install the application for the first time there is a notification that appears to choose to use root mode, or without root, or both.

Useless and dangerous:
* As mentioned above, you can grant root privileges through either `su` or `tsu` (preferred).
* Package manager is locked down for non-root use exclusively for obvious reasons: under root it messing up SELinux labels and ownership + certain chance of trashing whole `/data` partition (hopefully you understand that Termux packaging quality is lower than of Debian and other major distributions and everything may happen).
